model.py

```diff
@@ -0,0 +1,125 @@
+from flask_login import UserMixin
+from flask_sqlalchemy import SQLAlchemy
+
+db = SQLAlchemy()
+
+class User(UserMixin, db.Model):
+    id = db.Column(db.Integer, primary_key=True)
+    username = db.Column(db.String(80), unique=True, nullable=False)
+    email = db.Column(db.String(120), unique=True, nullable=False)
+    password = db.Column(db.String(255), nullable=False)
+    profile_picture = db.Column(db.String(255), nullable=True)
+    posts = db.relationship('Post', backref='author', lazy=True)
+    role = db.Column(db.Integer, db.ForeignKey('role.id'))
+    
+    following = db.relationship(
+        'Follow',
+        foreign_keys='Follow.follower_id',
+        lazy=True,
+        cascade='all, delete-orphan'
+    )
+
+    followers = db.relationship(
+        'Follow',
+        foreign_keys='Follow.followed_id',
+        lazy=True,
+        cascade='all, delete-orphan'
+    )
+    def follow(self, user):
+        if not self.is_following(user):
+            follow = Follow(follower=self, followed=user)
+            db.session.add(follow)
+
+    def unfollow(self, user):
+        follow = self.followings.filter_by(followed_id=user.id).first()
+        if follow:
+            db.session.delete(follow)
+
+    def is_following(self, user):
+        return self.following.filter_by(followed_id=user.id).first() is not None
+
+    def count_following(self):
+        return len(self.following)
+
+    def count_followers(self):
+        return len(self.followers)
+    
+    def count_likes_received(self):
+        return sum(post.count_likes() for post in self.posts)
+
+class Role(db.Model):
+    id = db.Column(db.Integer, primary_key=True)
+    name = db.Column(db.String(80), unique=True, nullable=False)
+    permissions = db.relationship('Permission', backref='role', lazy=True)
+
+class Permission(db.Model):
+    id = db.Column(db.Integer, primary_key=True)
+    name = db.Column(db.String(80), unique=True, nullable=False)
+    role_id = db.Column(db.Integer, db.ForeignKey('role.id'), nullable=False)
+
+
+class Post(db.Model):
+    id = db.Column(db.Integer, primary_key=True)
+    content = db.Column(db.Text, nullable=False)
+    image = db.Column(db.String(255), nullable=True)
+    user_id = db.Column(db.Integer, db.ForeignKey('user.id'))
+    likes = db.relationship('Liked_Post', backref='post', lazy=True, cascade='all, delete-orphan')
+    comments = db.relationship('Comment', backref='post', lazy=True, cascade='all, delete-orphan')
+    
+    def count_likes(self):
+        return len(self.likes)
+    def to_json(self):
+        return {
+            'id': self.id,
+            'content': self.content,
+            'image': self.image,
+            'user_id': self.user_id,
+            'likes': self.likes,
```

Comment:
likes and comments are relationship with this model, so doesn't required to return in json.